tests/test_flows.py

```diff
@@ -278,6 +279,10 @@ def run_tests(self, tests):
             self.check_error = 0
             try:
                 getattr(self, test_full)()
+            except AssertionError as ae:
+                self.fail(f"{ae}")
+            except Exception as e:
+                self.fail(f"Test has thrown an exception\n{traceback.format_exc()}")
```

Comment:
I don't think we should print the full traceback. Also, I think it's worth repeating the test name.
```suggestion
                self.fail("Test {} has thrown an exception\n{}".format(test, e))
```

(Incidentally, this will also fix the flake8 error because `e` is now used)

With that, there's also no real need to handle the AssertionError differently.

BTW, it would be nice to show in the commit message some examples of what the output will be when an exception is handled. Something like:
```
For example, when a non-existing TLV field is accessed, the following failure will be printed:
...
```
For reviewers, that's much easier to help decide if it is done properly.

Reply:
Ah, I totally missed that this was causing the flake8 error.

Reply:
I expect tests to throw AssertionError when they fail and decide to quit. However I consider other exceptions as unexpected run-time exceptions and therefore I conclude that the test code has failed, not the DUT. I think we definitely need the stack trace (in debug / verbose mode) so that we can have a chance to figure out what went wrong without having to debug the test.